server/src/main/kotlin/com/trib3/server/graphql/GraphQLWebSocket.kt

```diff
@@ -24,13 +118,22 @@ private val log = KotlinLogging.logger { }
  */
 open class GraphQLWebSocket(
     val graphQL: GraphQL?,
-    val objectMapper: ObjectMapper
+    val objectMapper: ObjectMapper,
+    val scheduler: Scheduler = Schedulers.io() // use the io scheduler by default
 ) : WebSocketAdapter() {
-    val objectWriter = objectMapper.writerWithDefaultPrettyPrinter()
+    val objectWriter = objectMapper.writerWithDefaultPrettyPrinter()!!
+
+    private val remoteLock = ReentrantLock()
 
     private val runningQueryLock = ReentrantLock()
     @Volatile
-    private var runningQuerySubscriber: GraphQLQuerySubscriber? = null
+    internal var runningQuerySubscriber: GraphQLQuerySubscriber? = null
+        private set
```

Comment:
whats `private set`?

Reply:
it makes the setter private even though the var is internal.  basically did this just for test coverage